src/server/server_family.cc

```diff
@@ -2079,25 +2083,29 @@ std::optional<ServerFamily::VersionBasename> ServerFamily::GetVersionAndBasename
     }
   }
 
-  string_view basename;
+  string sub_path;
+  string basename;
   if (args.size() == 2) {
-    basename = ArgS(args, 1);
+    fs::path path(ArgS(args, 1));
+    sub_path = path.parent_path().relative_path().string();
+    basename = path.filename().string();
```

Comment:
nit you probably mean `filename` and not `basename` which implies a dir

Reply:
Basename was used before, where you specified prefix for all saved files. There was no change in logic there, so i kept basename.